spyder/plugins/editor/panels/codefolding.py

```diff
@@ -213,11 +213,14 @@ def update_folding(self, ranges):
             current_lines = self.__compute_line_offsets(current_text, True)
 
             for line in past_folding_regions:
-                offset = prev_offsets[line]
-                new_offset = differ.diff_xIndex(diff, offset)
-                new_line = current_lines[new_offset]
-                if new_line in self.folding_regions:
-                    folding_status[new_line] = self.folding_status[line]
+                try:
+                    offset = prev_offsets[line]
+                    new_offset = differ.diff_xIndex(diff, offset)
+                    new_line = current_lines[new_offset]
+                    if new_line in self.folding_regions:
```

Comment:
Why not something like:
```python
                    offset = prev_offsets[line]
                    new_offset = differ.diff_xIndex(diff, offset)
                    new_line = current_lines[new_offset]
                    if new_line in self.folding_regions and line in self.folding_status[line]:
                        folding_status[new_line] = self.folding_status[line]
```

or

```python
                try:
                    if new_line in self.folding_regions: 
...
```

The problem line is only `folding_status[new_line] = self.folding_status[line]`